### PR TITLE
feat: handle partial state objects

### PR DESCRIPTION
**Pull request by Julusian**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Fix/Feature

* **What is the current behavior?** (You can also link to an open issue here)

The input states are expected to be 'complete' objects, according to the AtemState definition. If some expected keys are missing, that can result in TypeErrors being thrown.

* **What is the new behavior (if this is a feature change)?**

The library should be able to handle partial objects. When any keys are missing, the 'default' values should be used instead

* **Other information**:

This has not yet been tested, and could do with some work to test edge cases. It should be possible to automate this by taking complete state objects, deleting random keys and making sure a resolve doesnt error.

### CODE REVIEW

**Review by baltedewit — COMMENTED** [review]
In general this looks like it will make the library much more robust so 👍 

There are a couple of places that appear to introduce hard coded defaults, I would suggest moving those out to the defaults files. In addition to that there's some inconsistency if a property is missing: it will sometimes be assumed to be at default and sometimes it will fire an extra command

The case of `'previewInput' in newMixEffect && `input` in oldMixEffect` should be refactored to allow mixing and matching `previewInput` and `input` sometime but not for this PR.

File: src/resolvers/mixEffect.ts [hunk 1]
```diff
@@ -53,33 +59,33 @@ export function resolveMixEffectsState(
 					commands.push(new AtemCommands.AutoTransitionCommand(mixEffectId))
 				}
 			}
-		} else if ('previewInput' in oldMixEffect && 'previewInput' in newMixEffect) {
-			if (oldMixEffect.previewInput !== newMixEffect.previewInput) {
-				commands.push(new AtemCommands.PreviewInputCommand(mixEffectId, newMixEffect.previewInput))
+		} else if ((!oldMixEffect || 'previewInput' in oldMixEffect) && (!newMixEffect || 'previewInput' in newMixEffect)) {
```
[the remark this answers is not in the document]
Comment (Julusian) [reply src/resolvers/mixEffect.ts:62]: well, the idea I have been working with is that if one of the properties is unknown, then we assume we need to send a command to get it into a known state. But if they are both unknown, then we can leave it alone.
I dont know if this is the correct behaviour if the new state has it undefined, but its consistent with what we have been doing elsewhere in this for a while.

**Comment by Julusian** [comment]
Some tests have been written as part of #20 (they are dependent on further changes added there).
